File: web/src/lib/memberPicker.ts
/**
 * Pure filter/sort for the "invite a partner" member picker (plan Phase 3b
 * task): search the members directory by name, excluding the signed-in
 * member and anyone already confirmed on this session.
 */
import type { Member } from '@obc/shared';

export interface FilterPickableMembersOptions {
  selfId: string;
  /** memberIds already confirmed on this session (derived from the roster). */
  excludeMemberIds: Iterable<string>;
  query: string;
}

export function filterPickableMembers(members: Member[], opts: FilterPickableMembersOptions): Member[] {
  const exclude = new Set(opts.excludeMemberIds);
  const q = opts.query.trim().toLowerCase();
  return members
    .filter((m) => m.id !== opts.selfId && !exclude.has(m.id))
    .filter((m) => !q || `${m.firstName} ${m.lastName}`.toLowerCase().includes(q))
    .sort((a, b) => `${a.firstName} ${a.lastName}`.localeCompare(`${b.firstName} ${b.lastName}`));
}

File: ios/OBCDanceCard/Features/Common/MemberPicker.swift
//
//  MemberPicker.swift
//  Pure filter/sort for the "invite a partner" member picker — a 1:1 port of
//  `web/src/lib/memberPicker.ts`: search the members directory by name,
//  excluding the signed-in member and anyone already confirmed on this
//  session.
//

import Foundation

enum MemberPicker {
    static func filter(
        _ members: [Member],
        selfId: String,
        excludeMemberIds: Set<String>,
        query: String
    ) -> [Member] {
        let q = query.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        return members
            .filter { $0.id != selfId && !excludeMemberIds.contains($0.id) }
            .filter { q.isEmpty || $0.fullName.lowercased().contains(q) }
            .sorted { $0.fullName.localizedCaseInsensitiveCompare($1.fullName) == .orderedAscending }
    }
}
